### ISSUE TITLE
No such file or directory

### ISSUE THREAD

**Issue by ctlssr** [yoonwaiyan/stackskills-dl#23]
Hello. I am new to ruby, could you please help me out with the following issue? Thanks!
The first folder is being created, but that's about it.

youtube-dl is installed.

`ruby stackskills_dl.rb`
`Login Email: *****@gmail.com`
`Login password: **********`
`Login Successfully.`
`Downloading Course: ********`
`Downloading What_is_Python-`
`01. What_is_Python-`
`C:/Users/*****/Desktop/Programming/Resources/stackskills-dl-master/lib/lecture.rb:50:in """: No such file or directory - youtube-dl --restrict-filenames https://fast.wistia.net/embed/iframe/********* (Errno::ENOENT)`

**Comment by ctlssr** [comment]
```
ruby stackskills_dl.rb -cs
Login Email: *****@gmail.com
Login password: **********
Login Successfully.
Finding s from your list of courses
Downloading Course: StackSkills
```

Then it simply stops.

**Comment by ctlssr** [comment]
Well, tried it in Git bash and got back to square one: SSL error.

`SSL_connect SYSCALL returned=5 errno=0 state=SSLv2/v3 read server hello A (OpenSSL::SSL::SSLError)`

I guess "Windows not supported" means Windows not supported. You can aswell close the issue :(

**Comment by yoonwaiyan** [comment]
Hi there, sorry I don't have a Windows machine with me to solve this problem, but if there's any pull request that solves this problem I'll be glad to merge it. 

I've looked into the problem and trying to solve it  by the latest commit, can you please have a try on the latest pull and update me whether it's working from your side?

**Comment by ctlssr** [comment]
Same result. I have youtube-dl, do I need wget?

`No such file or directory - youtube-dl --restrict-filenames https://fast.wistia.net/embed/iframe/********* (Errno::ENOENT)`

**Comment by yoonwaiyan** [comment]
Can you verify youtube-dl is running well from the directory using git bash? i.e execute `youtube-dl` from the directory. wget is necessary if you have course that has any direct download link of the video but it's not related to the error message for now.

**Comment by ctlssr** [comment]
I think the problem is not with youtube-dl.

`puts youtube-dl #{video_url}` works like a charm with the video_url that the code gets

I think it's mkchdir that's the problem, I'll look into it later.

**Comment by yoonwaiyan** [comment]
Hi there, any update of this issue?

**Comment by boettges** [comment]
@ctlssr you are missing wget 👌

**Comment by ctlssr** [comment]
@boettges wget is installed. I'm on Windows.

**Comment by yoonwaiyan** [comment]
Hi there, for the issue as shown in OP, I got a hint on this issue while building up a feature, but need some clarification from you first before confirming the problem. 

1. May I know what's the lecture name in StackSkills as shown in OP? What is Python is the full lecture name?
2. If there's any other courses, please run the script to download another course. Is this problem persists?

While for the second post, I think the problem happens because the script used to accept `-cs` but the option is invalid thus i did a fix in a2155b8c66deb24fe1b315def3d49d7341adadcb . If you've face this problem, the correct command to download specific course is either `-c` or `-s`.

**Comment by yoonwaiyan** [comment]
Hi there, I'll be closing this issue as the issues as reported are possibly solved through other issues and fixes. Feel free to re-open the issue if problem persists, thanks.
